packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -573,6 +573,11 @@ export class Function extends FunctionBase {
    */
   public readonly deadLetterQueue?: sqs.IQueue;
 
+  /**
+   * The timeout configured for this lambda.
+   */
+  public readonly timeout?: Duration;
```

Comment:
For our needs, an optional value is OK because we could throw an exception if the timeout was not defined.

A non-optional value would be a better API but this value is not guaranteed to be in the CloudFormation template. As such, if the default were changed in the future it might lead to CDK reporting a timeout value different to the one deployed to Lambda.

However, I'm happy to make that change if you're OK with this caveat. 👍

Reply:
Ok, fair enough. Let's leave it as is.